lib/rules/await-async-events.ts

```diff
@@ -153,6 +154,13 @@ export default createTestingLibraryRule<Options, MessageIds>({
 										findClosestFunctionExpressionNode(node);
 
 									if (functionExpression) {
+										const deepestCalleeIdentifier = getDeepestIdentifierNode(
+											functionExpression.parent
+										);
+										if (deepestCalleeIdentifier?.name === 'forEach') {
```

Comment:
(not a maintainer of this package) I would add these commonly mis-used iterator methods
```suggestion
										if (['reduce', 'forEach', 'map'].includes(deepestCalleeIdentifier?.name)) {
```

Reply:
disregard this comment, it will require more complexity (understand if something has been returned)